HW2/P2/P2.txt

```diff
@@ -0,0 +1 @@
+It's interesting that serial is by far the fastest. When I run fine grained it is around 5 seconds, the medium grained locking ranges from 6 seconds to 10 based on correlation and lock size. As N (the number of items locked at a time) increases, the program slows down. This makes sense because the concurrency is greatest with more fine grained locking. The fastest time for correlated seemed to occur at N = 10. This makes sense because the source and destination are no more than 10 apart, so you avoid having to manage as many locks as fine grained while still limiting the time one thread waits for another lock. 
```

Comment:
Good answer, I got similar values - however, I saw better performance for larger values of N for correlated data